spec/system/admin/products_v3/products_spec.rb

```diff
@@ -926,6 +926,45 @@ def expect_other_columns_visible
       end
     end
 
+    context 'When trying to save an invalid variant with Stock value ' do
+      let(:new_variant_row) { find_field("Name", placeholder: "Apples", with: "").ancestor("tr") }
+
+      before do
+        visit admin_products_url
+        click_on "New variant"
+      end
+
+      it 'displays the correct value afterwards for On Hand' do
+        within new_variant_row do
+          fill_in "Name", with: "Large box"
+          click_on "On Hand"
+          fill_in "On Hand", with: "19"
+        end
+
+        click_button "Save changes"
+
+        expect(page).to have_content "Please review the errors and try again"
+        within row_containing_name("Large box") do
+          expect(page).to have_content "19"
+        end
+      end
+
+      it 'displays the correct value afterwards for On demand' do
+        within new_variant_row do
+          fill_in "Name", with: "Large box"
+          click_on "On Hand"
+          check "On demand"
+        end
+
+        click_button "Save changes"
+
+        expect(page).to have_content "Please review the errors and try again"
+        within row_containing_name("Large box") do
+          expect(page).to have_content "On demand"
+        end
+      end
```

Comment:
Great to have this tested! :100: 

Since system specs are slow, I would combine these two examples. You can fill both values, hit save once and then assert on both values.

Reply:
I have refactored the spec in this way :)